File: lab3/Tipper/app/src/main/java/com/example/tipper/BMICalculatorActivity.java
/**
 * BMI Calculator
 * Simple Body mass index calculator in Java
 *
 * @author  Sebastian Lewandowski
 * @version 1.0
 */

package com.example.tipper;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.text.Editable; // for EditText event handling
import android.text.TextWatcher; // EditText listener
import android.view.View;
import android.widget.Button;
import android.widget.EditText; // for bill amount input
import android.widget.SeekBar; // for changing the tip percentage
import android.widget.SeekBar.OnSeekBarChangeListener; // SeekBar listener
import android.widget.TextView; // for displaying text

import java.text.NumberFormat; // for currency formatting

public class BMICalculatorActivity extends AppCompatActivity {

    // weight formatter
    private static final NumberFormat numberFormat =
            NumberFormat.getNumberInstance();

    private double weightValue = 0.0; // weight value entered by the user
    private double heightValue = 1.75; // initial height value
    private double BMIValue =0.0;
    private TextView weightTextView; // shows formatted weight
    private TextView heightTextView; // shows height value set up by user
    private TextView BMITextView; // shows calculated BMI value
    private TextView SummaryTextView;
    private Button btnOpenChartActivity;

    // called when the activity is first created
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState); // call superclass onCreate
        setContentView(R.layout.activity_bmi_calc); // inflate the GUI

        // get references to programmatically manipulated TextViews
        weightTextView = (TextView) findViewById(R.id.weightTextView);
        heightTextView = (TextView) findViewById(R.id.heightTextView);
        BMITextView = (TextView) findViewById(R.id.BMITextView);
        SummaryTextView = (TextView) findViewById(R.id.SummaryTextView);
        btnOpenChartActivity = (Button) findViewById(R.id.btnOpenChartActivity);

        BMITextView.setText(numberFormat.format(0));

        // set weightEditText's TextWatcher
        EditText weightEditText =
                (EditText) findViewById(R.id.weightEditText);
        weightEditText.addTextChangedListener(weightEditTextWatcher);

        // set heightSeekBar's OnSeekBarChangeListener
        SeekBar heightSeekBar =
                (SeekBar) findViewById(R.id.heightSeekBar);
        heightSeekBar.setOnSeekBarChangeListener(seekBarListener);

        btnOpenChartActivity.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent i = new Intent(BMICalculatorActivity.this, ChartActivity.class);
                OpenChartActivityOnButtonClick();
                if (BMIValue != 0.0) {
                    String bmiValueToString = Double.toString(BMIValue);
                    i.putExtra("bmiValueToString", bmiValueToString);
                    startActivity(i);
                }
            }
        });
    }

    // calculate and display BMI value
    private void calculate() {
        // format percent and display in percentTextView
        heightTextView.setText(numberFormat.format(heightValue));

        // calculate the BMI value
        double denominator_pow = Math.pow(heightValue,2);
        BMIValue = weightValue / denominator_pow;

        // display formatted BMI value
        BMITextView.setText(numberFormat.format(BMIValue));

        // print summary for calculated BMI value
        if(BMIValue < 16.0){
            SummaryTextView.setText("wygłodzenie");
            SummaryTextView.setTextColor(Color.rgb(153, 0, 0));
        }
        else if(BMIValue >= 16.0 && BMIValue < 17.0){
            SummaryTextView.setText("wychudzenie");
            SummaryTextView.setTextColor(Color.rgb(255, 0, 0));
        }
        else if(BMIValue >= 17.0 && BMIValue < 18.5){
            SummaryTextView.setText("niedowaga");
            SummaryTextView.setTextColor(Color.rgb(230, 230, 0));
        }
        else if(BMIValue >= 18.5 && BMIValue < 25.0){
            SummaryTextView.setText("waga prawidłowa");
            SummaryTextView.setTextColor(Color.rgb(0, 255, 0));
        }
        else if(BMIValue >= 25.0 && BMIValue < 30.0){
            SummaryTextView.setText("nadwaga");
            SummaryTextView.setTextColor(Color.rgb(255, 128, 0));
        }
        else if(BMIValue >= 30.0 && BMIValue < 35.0){
            SummaryTextView.setText("otyłość I stopnia");
            SummaryTextView.setTextColor(Color.rgb(255, 0, 0));
        }
        else if(BMIValue >= 35.0 && BMIValue < 40.0){
            SummaryTextView.setText("otyłość II stopnia");
            SummaryTextView.setTextColor(Color.rgb(102, 0, 0));
        }
        else if(BMIValue >= 40.0){
            SummaryTextView.setText("otyłość III stopnia");
            SummaryTextView.setTextColor(Color.rgb(51, 0, 0));
        }
        else
            SummaryTextView.setText("");
    }

    // listener object for the SeekBar's progress changed events
    private final OnSeekBarChangeListener seekBarListener =
            new OnSeekBarChangeListener() {
                // update height, then call calculate
                @Override
                public void onProgressChanged(SeekBar seekBar, int progress,
                                              boolean fromUser) {
                    heightValue = progress / 100.0 ; // set height based on progress
                    calculate(); // calculate and display BMI
                }

                @Override
                public void onStartTrackingTouch(SeekBar seekBar) { }

                @Override
                public void onStopTrackingTouch(SeekBar seekBar) { }
            };

    // listener object for the EditText's text-changed events
    private final TextWatcher weightEditTextWatcher = new TextWatcher() {
        // called when the user modifies the bill amount
        @Override
        public void onTextChanged(CharSequence s, int start,
                                  int before, int count) {

            try { // get weight and display formatted value
                weightValue = Double.parseDouble(s.toString()) / 10.0;
                weightTextView.setText(numberFormat.format(weightValue));
            }
            catch (NumberFormatException e) { // if s is empty or non-numeric
                weightTextView.setText("");
                weightValue = 0.0;
            }

            calculate(); // update the BMI TextView
        }

        @Override
        public void afterTextChanged(Editable s) { }

        @Override
        public void beforeTextChanged(
                CharSequence s, int start, int count, int after) { }
    };

    private void OpenChartActivityOnButtonClick() {
        Intent openChartActivity = new Intent(this, ChartActivity.class);
        if(BMIValue != 0.0)
            startActivity(openChartActivity);
    }
}


/*************************************************************************
 * (C) Copyright 1992-2016 by Deitel & Associates, Inc. and               *
 * Pearson Education, Inc. All Rights Reserved.                           *
 *                                                                        *
 * DISCLAIMER: The authors and publisher of this book have used their     *
 * best efforts in preparing the book. These efforts include the          *
 * development, research, and testing of the theories and programs        *
 * to determine their effectiveness. The authors and publisher make       *
 * no warranty of any kind, expressed or implied, with regard to these    *
 * programs or to the documentation contained in these books. The authors *
 * and publisher shall not be liable in any event for incidental or       *
 * consequential damages in connection with, or arising out of, the       *
 * furnishing, performance, or use of these programs.                     *
 *************************************************************************/

File: lab4/BMI_App/app/src/main/java/com/example/bmiapp/BMICalculatorActivity.kt
package com.example.bmiapp

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.widget.Button
import android.widget.EditText
import android.widget.SeekBar
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import java.text.DecimalFormat
import java.text.NumberFormat
import kotlin.math.pow


class BMICalculatorActivity : AppCompatActivity() {

    private var weightValue = 0.0
    private var heightValue = 1.75
    private var bMIValue = 0.0
    private val numberFormat = NumberFormat.getNumberInstance()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_bmi_calc)

        val weightEditText = findViewById<EditText>(R.id.weightEditText)
        val heightV = findViewById<TextView>(R.id.bmi_height_value)
        val heightSeekBar = findViewById<SeekBar>(R.id.heightSeekBar)
        val btnChartActivity = findViewById<Button>(R.id.btnOpenChartActivity)

        weightEditText.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
            }
            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
                if (!p0.isNullOrEmpty()) {
                    weightValue = p0.toString().toDouble()
                    calculate()
                }
            }

            override fun afterTextChanged(p0: Editable?) {
            }
        })

        heightSeekBar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
            override fun onProgressChanged(p0: SeekBar?, p1: Int, p2: Boolean) {
                heightValue = p1.toDouble().div(100)
                heightV.text = heightValue.toString()
                calculate()
            }

            override fun onStartTrackingTouch(p0: SeekBar?) {
            }

            override fun onStopTrackingTouch(p0: SeekBar?) {
            }
        })

        btnChartActivity.setOnClickListener{
            val chartActivity = Intent(this, ChartActivity::class.java)
            if (bMIValue > 0.0)
                startActivity(chartActivity)
        }

    }

    private fun calculate() {
        val bmiValueDisplay = findViewById<TextView>(R.id.bmi_value)
        val bmiSummary = findViewById<TextView>(R.id.bmi_summary)
        if (weightValue > 0.0) {
            val denominatorPow = heightValue.pow(2)
            bMIValue = weightValue / denominatorPow
            bmiValueDisplay.text = numberFormat.format(bMIValue).toString()
                if(bMIValue < 16.0){
                    bmiSummary.text ="wygłodzenie"
                    bmiSummary.setTextColor(Color.rgb(153, 0, 0));
                }
                else if(bMIValue >= 16.0 && bMIValue < 17.0){
                    bmiSummary.text = "wychudzenie"
                    bmiSummary.setTextColor(Color.rgb(255, 0, 0));
                }
                else if(bMIValue >= 17.0 && bMIValue < 18.5){
                    bmiSummary.text ="niedowaga"
                    bmiSummary.setTextColor(Color.rgb(230, 230, 0));
                }
                else if(bMIValue >= 18.5 && bMIValue < 25.0){
                    bmiSummary.text ="waga prawidłowa"
                    bmiSummary.setTextColor(Color.rgb(0, 255, 0));
                }
                else if(bMIValue >= 25.0 && bMIValue < 30.0){
                    bmiSummary.text ="nadwaga"
                    bmiSummary.setTextColor(Color.rgb(255, 128, 0));
                }
                else if(bMIValue >= 30.0 && bMIValue < 35.0){
                    bmiSummary.text ="otyłość I stopnia"
                    bmiSummary.setTextColor(Color.rgb(255, 0, 0));
                }
                else if(bMIValue >= 35.0 && bMIValue < 40.0){
                    bmiSummary.text ="otyłość II stopnia"
                    bmiSummary.setTextColor(Color.rgb(102, 0, 0));
                }
                else if(bMIValue >= 40.0){
                    bmiSummary.text ="otyłość III stopnia"
                    bmiSummary.setTextColor(Color.rgb(51, 0, 0));
                }
                else
                    bmiSummary.text =""
        }
    }
}



